src/Avalonia.Input/TextInput/InputMethodManager.cs

```diff
@@ -75,27 +75,35 @@ public void SetFocusedElement(IInputElement? element)
             if(_focusedElement == element)
                 return;
             _focusedElement = element;
-            
-            var inputMethod = (element?.VisualRoot as ITextInputMethodRoot)?.InputMethod;
+
+            var root = element?.VisualRoot as ITextInputMethodRoot;
+
+            NotifyInputMethodUpdated(root);
+        }
+
+        public void NotifyInputMethodUpdated(ITextInputMethodRoot? root)
```

Comment:
Why exactly do we need this? Is it needed to handle keyboard layout switch between basic and IME-enabled languages?

Reply:
I'm not sure if that should be handled on the xplat layer though. I'd expect that state to be tracked by the platform code that encapsulates such switches.

Reply:
Not sure how other users do this, but for me I heavily rely on win+space to switch between en/zh on the fly.
The ime manager queries client capabilities when it gets focus (or init?), but doesn't handle IME switch when the control is focused.

Reply:
Well, I'd expect the WindowImpl.cs to always have an active `Imm32InputMethod` instance (just like X11Window does) that maintains its state rather than going through the requery process. Your approach also adds more ties between input method handling and toplevels while we are kinda planning to add built-in OSK support.

Reply:
`WindowImpl` always have that. `InputMethodManager` doesn't.

Reply:
The initial idea was that InputMethodManager synchronizes the control with the native window implementation. The native window implementation is supposed to keep track of what was previously passed to it rather than requery that information on demand.

Reply:
I see. Maybe it's possible to put the mutable states in IMM32InputMethod.